src/discovery.ts

```diff
@@ -125,6 +125,10 @@ export const bestPriceDiscovery: Discovery = async (
         }>).value
     );
 
+  if (pricesWithClients.length === 0) {
+    throw new Error('No valid trade order has been found');
```

Comment:
```suggestion
    throw new Error('No provider available or enough liquidity for the requested amount');
```

Reply:
All my orders were rejected with `provided amount is too low`, your error message doesn't include that case.